<a href="https://colab.research.google.com/github/mw1849/nyu-python-public-policy/blob/master/HW2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd
from google.colab import drive

In [0]:
drive.mount('/content/drive')


Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code

Enter your authorization code:
··········
Mounted at /content/drive


In [0]:
df = pd.read_csv('/content/drive/My Drive/Data for Python/cleaned_311_data.csv', header='infer')

/usr/local/lib/python3.6/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (9,18,21,32,35) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [0]:
df.groupby('open_data_channel_type').size().nlargest().reset_index(name='count')

open_data_channel_type    count
0                  PHONE  1463204
1                 ONLINE   589865
2                UNKNOWN   417916
3                 MOBILE   368759
4                  OTHER    19267

Q1. Most people use the phone to submit their complaints to 311. 

In [0]:
sub_method = df.groupby(['open_data_channel_type', 'complaint_type']).size().reset_index(name='count')

In [0]:
sub_method.sort_values(['complaint_type'], ascending=[True])

open_data_channel_type           complaint_type  count
223                  PHONE                APPLIANCE  12002
40                  ONLINE                APPLIANCE   2627
0                   MOBILE        Abandoned Vehicle   2484
224                  PHONE        Abandoned Vehicle   5069
41                  ONLINE        Abandoned Vehicle   1891
..                     ...                      ...    ...
194                 ONLINE             Window Guard     17
413                  PHONE             Window Guard     13
518                UNKNOWN             Window Guard      1
195                 ONLINE  X-Ray Machine/Equipment      5
414                  PHONE  X-Ray Machine/Equipment      4

[519 rows x 3 columns]

Looking at the data, it seems that some complaints are made more frequently depending on when and where people would usually make the complaints. For example, when the complaint is for an abandoned vehicle, people usually submit the complaint by phone/mobile (5069 complaints) because they may usually be outside when making the complaint. However, for complaints about X-Ray Machine/Equipment, this would usually be done inside, with access to computers, so complaints would be made most often through an website (online) submission.

In [0]:
sub_method.sort_values(['count','complaint_type'], ascending=[False,True]).drop_duplicates(['complaint_type'])

open_data_channel_type                       complaint_type   count
367                  PHONE  Request Large Bulky Item Collection  177175
303                  PHONE                       HEAT/HOT WATER  128002
343                  PHONE                  Noise - Residential   95407
244                  PHONE                     Blocked Driveway   95206
501                UNKNOWN               Street Light Condition   77755
..                     ...                                  ...     ...
245                  PHONE                        Bottled Water       5
83                  ONLINE            Fire Alarm - Reinspection       5
454                UNKNOWN                 Forensic Engineering       5
152                 ONLINE                            Rangehood       5
195                 ONLINE              X-Ray Machine/Equipment       5

[229 rows x 3 columns]